File: AdjacencyList.java
import java.util.LinkedList;
import java.util.*;

class AdjacencyList{
	ArrayList<LinkedList<Integer>> adjList;
	int v;

	AdjacencyList(int z){
		v=z;
		adjList=new ArrayList<LinkedList<Integer>>();

		for(int i=0;i<v;i++){
			adjList.add(new LinkedList<Integer>());
		}
	}

	void add(int src, int dest){
		adjList.get(src).add(dest);
		adjList.get(dest).add(src);
//		System.out.println(adjList);
	}

	void printGraph(){
		for(int i=0;i<v;i++){
			System.out.print(i+" --> ");
			for(int z:adjList.get(i)){
				System.out.print(z+" ");				
			}
			System.out.println();
		}
	}

	public static void main(String[] args){
		AdjacencyList graph=new AdjacencyList(5);

		graph.add(0,1);
		graph.add(0,4);
		graph.add(1,2);
		graph.add(1,3);
		graph.add(1,4);
		graph.add(2,3);
		graph.add(3,4);
		
		graph.printGraph();

	}

}

File: AdjacencyList.py
class Node:
	def __init__(self,data):
		self.vertex=data
		self.next=None

class AdjList:
	def __init__(self,vtx):
		self.V=vtx
		self.graph=[None]*self.V

	def addNode(self,src,dest):
		nd=Node(src)
		nd.next=self.graph[dest]
		self.graph[dest]=nd

		nd2=Node(dest)
		nd2.next=self.graph[src]
		self.graph[src]=nd2

	def print_graph(self):
		for i in range(self.V):
			print("{}".format(i))
			temp=self.graph[i]
			while temp:
				print("{}".format(temp.vertex))
				temp=temp.next
			print(" \n")

llist=AdjList(5)
llist.addNode(0,1)
llist.addNode(0,4)
llist.addNode(1,4)
llist.addNode(1,2)
llist.addNode(2,3)
llist.addNode(1,3)
llist.addNode(3,4)
llist.print_graph()
